File: src/main/java/responses/user/Counts.java
package responses.user;

public class Counts {
    private final int pins;
    private final int following;
    private final int followers;
    private final int boards;
    private final int likes;

    public Counts(int pins, int following, int followers, int boards, int likes) {
        this.pins = pins;
        this.following = following;
        this.followers = followers;
        this.boards = boards;
        this.likes = likes;
    }

    public int getPins() {
        return pins;
    }

    public int getFollowing() {
        return following;
    }

    public int getFollowers() {
        return followers;
    }

    public int getBoards() {
        return boards;
    }

    public int getLikes() {
        return likes;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;

        Counts counts = (Counts) o;

        if (pins != counts.pins) return false;
        if (following != counts.following) return false;
        if (followers != counts.followers) return false;
        if (boards != counts.boards) return false;
        return likes == counts.likes;

    }

    @Override
    public int hashCode() {
        int result = pins;
        result = 31 * result + following;
        result = 31 * result + followers;
        result = 31 * result + boards;
        result = 31 * result + likes;
        return result;
    }

    @Override
    public String toString() {
        return "Counts{" +
                "pins=" + pins +
                ", following=" + following +
                ", followers=" + followers +
                ", boards=" + boards +
                ", likes=" + likes +
                '}';
    }
}

File: src/main/java/com/chrisdempewolf/pinterest/responses/user/Counts.kt
package com.chrisdempewolf.pinterest.responses.user

class Counts(val pins: Int, val following: Int, val followers: Int, val boards: Int, val likes: Int) {

    override fun equals(o: Any?): Boolean {
        if (this === o) return true
        if (o == null || javaClass != o.javaClass) return false

        val counts = o as Counts?

        if (pins != counts!!.pins) return false
        if (following != counts.following) return false
        if (followers != counts.followers) return false
        if (boards != counts.boards) return false
        return likes == counts.likes

    }

    override fun hashCode(): Int {
        var result = pins
        result = 31 * result + following
        result = 31 * result + followers
        result = 31 * result + boards
        result = 31 * result + likes
        return result
    }

    override fun toString(): String {
        return "Counts{" +
                "pins=" + pins +
                ", following=" + following +
                ", followers=" + followers +
                ", boards=" + boards +
                ", likes=" + likes +
                '}'
    }
}
